emcc.py

```diff
@@ -2534,6 +2535,16 @@ def phase_final_emitting(options, target, wasm_target, memfile):
     shared.JS.handle_license(final_js)
     shared.run_process([shared.PYTHON, shared.path_from_root('tools', 'hacky_postprocess_around_closure_limitations.py'), final_js])
 
+  # Unmangle previously mangled `import.meta` references in both main code and libraries.
+  # See also: `preprocess` in parseTools.js.
+  if settings.EXPORT_ES6 and settings.USE_ES6_IMPORT_META:
+    with open(final_js, 'r+') as f:
+      src = f.read()
+      src = src.replace('EMSCRIPTEN$IMPORT$META', 'import.meta')
```

Comment:
if you use `do_replace` it will assert that the pattern was found.

Reply:
Ah nice, TIL. I'm trying to think if there are situations where code wouldn't use `import.meta` and we wouldn't want such assertions... maybe when using asm.js output?

Reply:
Yeah, just checked, in case of asm.js there is no `import.meta` instances and correspondingly `.replace` won't find anything. There are probably more scenarios where this is true, so for simplicity I'll just keep `.replace` here.